examples/tests/app_loader/button-press-loading/main.c

```diff
@@ -0,0 +1,287 @@
+*
+* 1. Callback to let us know when the capsule is done writing data to flash
+* 2. Set button callback to initiate the dynamic app load process on pressing button 1 (on nrf52840dk)
+*
+******************************************************************************************************/
+
+static void app_setup_done_callback(__attribute__((unused)) int   arg0,
+                                    __attribute__((unused)) int   arg1,
+                                    __attribute__((unused)) int   arg2,
+                                    __attribute__((unused)) void* ud) {
+  setup_done = true;
+}
+
+static void app_write_done_callback(__attribute__((unused)) int   arg0,
+                                    __attribute__((unused)) int   arg1,
+                                    __attribute__((unused)) int   arg2,
+                                    __attribute__((unused)) void* ud) {
+  write_done = true;
+}
+
+static void app_finalize_done_callback(__attribute__((unused)) int   arg0,
+                                       __attribute__((unused)) int   arg1,
+                                       __attribute__((unused)) int   arg2,
+                                       __attribute__((unused)) void* ud) {
+  finalize_done = true;
+}
+
+static void app_load_done_callback(int                           arg0,
+                                   __attribute__((unused)) int   arg1,
+                                   __attribute__((unused)) int   arg2,
+                                   __attribute__((unused)) void* ud) {
+
+  if (arg0 != RETURNCODE_SUCCESS) {
+    printf("[Error] Process creation failed: %d.\n", arg0);
+  } else {
+    printf("[Success] Process created successfully.\n");
+  }
+  load_done = true;
+}
+
+// Callback for button presses.
+static void button_callback(__attribute__ ((unused)) returncode_t retval, int btn_num, bool pressed) {
+  // Callback for button presses.
+  // val: 1 if pressed, 0 if depressed
+  if (pressed == 1 && !app_load) {
+    // Note: this variable is introduced
+    // because the current alarm upcall implementation
+    // results in panic when a button is pressed within
+    // the debounce period if the debounce period is long.
+    // Setting the debounce interval to 100ms seems to work
+    // but setting it to 200ms and rapidly clicking buttons
+    // leads to the kernel panicking.
+    app_load = true;
+    libtocksync_alarm_delay_ms(200); // debounce
+
+    if (pressed == 1) {
+      app_name = NULL;
+      app_data = NULL;
+      app_size = 0;
+
+      switch (btn_num) {
+        case BUTTON1:
+          app_name = "blink";
+          app_data = APP_BLINK;
+          app_size = sizeof(APP_BLINK);
+          break;
+        case BUTTON2:
+          app_name = "adc";
+          app_data = APP_ADC;
+          app_size = sizeof(APP_ADC);
+          break;
+        default:
+          printf("[Log] Unsupported Button.\n");
+          app_load = false;
+          return;
+      }
+    }
```

Comment:
```suggestion
    switch (btn_num) {
      case BUTTON1:
        app_name = "blink";
        app_data = APP_BLINK;
        app_size = sizeof(APP_BLINK);
        app_load = true;
        break;
      case BUTTON2:
        app_name = "adc";
        app_data = APP_ADC;
        app_size = sizeof(APP_ADC);
        app_load = true;
        break;
      default:
        printf("[Log] Unsupported Button.\n");
        return;
    }
```

I think this can still be way simpler and remove the weird debounce thing if you just treat `app_load` as a "already busy, ignore button" flag.

Reply:
yeah i forgot to push that change lol

Reply:
ok i think i remember why i did not push it. it was because in the unsupported case, the debounce still exists and prints logs the unsupported button case multiple times. i found it to be quite ugly

Reply:
I might push for this change still. As-written here, for the "pushed a valid button case", the bounces are dropped silently. An extra print or two for the invalid button case isn't really something that users will see often, and I think having a simpler app is more valuable.